x-pack/plugins/aiops/public/components/log_categorization/log_categorization_for_embeddable/log_categorization_for_embeddable.tsx

```diff
@@ -332,140 +285,94 @@ export const LogCategorizationEmbeddable: FC<LogCategorizationEmbeddableProps> =
       setLoading(false);
     }
   }, [
+    cancelRequest,
     dataView,
-    loading,
-    selectedField,
     earliest,
+    embeddingOrigin,
+    fieldName,
+    getMinimumTimeRange,
+    intervalMs,
     latest,
+    loading,
     minimumTimeRangeOption,
-    cancelRequest,
-    getMinimumTimeRange,
-    searchQuery,
-    runValidateFieldRequest,
-    embeddingOrigin,
     runCategorizeRequest,
-    intervalMs,
+    runValidateFieldRequest,
+    searchQuery,
     toasts,
   ]);
 
+  useEffect(
+    function setOnChange() {
+      if (typeof onChange === 'function') {
+        onChange(data?.categories ?? []);
+      }
+      if (data !== null) {
```

Comment:
Defined data doesn't always imply that the visualisation has been rendered. Did you test the PDF report?

Reply:
This is a good spot. I didn't realise it was being used in that way and indeed the PDF export is catching the component too early.

Reply:
I've removed this and made changes to improve the reporting of the rendering being complete in https://github.com/elastic/kibana/pull/186539/commits/be39b67b5884076c5fba701574af17572423dede
However it isn't catching the chart being rendered, only the loading message. I've tested a few other embeddables and they are also doing this, so I think there is a deeper problem with the PDF export.